numpy 1.23.0rc1 changes `RuntimeWarning` for `0./0.` giving `nan` from `true_divide` to `divide`

Today (2022-05-26) NumPy released [`v1.23.0rc1`](https://pypi.org/project/numpy/1.23.0rc1/) (congratulations to the NumPy team! :rocket:) though the HEAD of dependencies tests caught a breaking change in `tests/test_infer.py` where `test_hypotest_return_calculator[True-True-True-toybased-kwargs1-ToyCalculator]`

https://github.com/scikit-hep/pyhf/blob/ea3249a0767d4fbd1c269f3147910e4a47314393/tests/test_infer.py#L176-L195

hit

```pytb
>       CLs = tensorlib.astensor(CLsb / CLb)
E       RuntimeWarning: invalid value encountered in divide

CLb        = array(0.)
CLsb       = array(0.)
_          = <pyhf.optimize.scipy_optimizer object at 0x7f0df414ed40>
bkg_only_distribution = <pyhf.infer.calculators.EmpiricalDistribution object at 0x7f0df42e0be0>
self       = <pyhf.infer.calculators.ToyCalculator object at 0x7f0df419e2c0>
sig_plus_bkg_distribution = <pyhf.infer.calculators.EmpiricalDistribution object at 0x7f0df4330520>
tensorlib  = <pyhf.tensor.numpy_backend.numpy_backend object at 0x7f0df427d1c0>
teststat   = array(3.93824492)
```

So in `numpy` `v1.23.0+` having `0./0.` will now raise a `RuntimeWarning`

```console
$ docker run --rm -ti python:3.10 /bin/bash
root@8b7036f32997:/# python -m venv venv && . venv/bin/activate
(venv) root@8b7036f32997:/# python -m pip --quiet install --upgrade pip setuptools wheel
(venv) root@8b7036f32997:/# python -m pip install --pre numpy
Collecting numpy
  Downloading numpy-1.23.0rc1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.0/17.0 MB 21.4 MB/s eta 0:00:00
Installing collected packages: numpy
Successfully installed numpy-1.23.0rc1
(venv) root@8b7036f32997:/# python
Python 3.10.4 (main, May 11 2022, 07:15:55) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> a = np.asarray([0.])
>>> b = np.asarray([0.])
>>> a/b
<stdin>:1: RuntimeWarning: invalid value encountered in divide
array([nan])
>>> exit()
(venv) root@8b7036f32997:/# 
```

This is fine, and the only reason that we're hitting this as an error is that we've turned on all warnings as errors in CI. But this is something to note in general.

Actually this is super small, as it is just a change from `true_divide` to `divide` as in `numpy` `v1.22.4`

```python
>>> import numpy as np
>>> np.__version__
'1.22.4'
>>> from importlib.metadata import version  # or doing it the way people seem to want these days
>>> version("numpy")
'1.22.4'
>>> a = np.asarray([0.])
>>> b = np.asarray([0.])
>>> a/b
<stdin>:1: RuntimeWarning: invalid value encountered in true_divide
array([nan])
```

So the "fix" is easy. Just regex

https://github.com/scikit-hep/pyhf/blob/ea3249a0767d4fbd1c269f3147910e4a47314393/pyproject.toml#L90

to accept `true_divide` or `divide`.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -87,7 +87,7 @@ filterwarnings = [
     'ignore:invalid value encountered in add:RuntimeWarning',  #FIXME
     "ignore:In future, it will be an error for 'np.bool_' scalars to be interpreted as an index:DeprecationWarning",  #FIXME: tests/test_tensor.py::test_pdf_eval[pytorch]
     'ignore:Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with:UserWarning',  #FIXME: tests/test_optim.py::test_minimize[no_grad-scipy-pytorch-no_stitch]
-    'ignore:divide by zero encountered in true_divide:RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
+    'ignore:divide by zero encountered in (?:true_|divide):RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
     'ignore:[A-Z]+ is deprecated and will be removed in Pillow 10:DeprecationWarning',  # keras
 ]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -83,11 +83,11 @@ filterwarnings = [
     'ignore:the `interpolation=` argument to percentile was renamed to `method=`, which has additional options:DeprecationWarning',  # Issue #1772
     "ignore:The interpolation= argument to 'quantile' is deprecated. Use 'method=' instead:DeprecationWarning",  # Issue #1772
     'ignore: Exception ignored in:pytest.PytestUnraisableExceptionWarning',  #FIXME: Exception ignored in: <_io.FileIO [closed]>
-    'ignore:invalid value encountered in (?:true_|divide):RuntimeWarning',  #FIXME
+    'ignore:invalid value encountered in *divide:RuntimeWarning',  #FIXME
     'ignore:invalid value encountered in add:RuntimeWarning',  #FIXME
     "ignore:In future, it will be an error for 'np.bool_' scalars to be interpreted as an index:DeprecationWarning",  #FIXME: tests/test_tensor.py::test_pdf_eval[pytorch]
     'ignore:Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with:UserWarning',  #FIXME: tests/test_optim.py::test_minimize[no_grad-scipy-pytorch-no_stitch]
-    'ignore:divide by zero encountered in (?:true_|divide):RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
+    'ignore:divide by zero encountered in *divide:RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
     'ignore:[A-Z]+ is deprecated and will be removed in Pillow 10:DeprecationWarning',  # keras
 ]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -83,7 +83,7 @@ filterwarnings = [
     'ignore:the `interpolation=` argument to percentile was renamed to `method=`, which has additional options:DeprecationWarning',  # Issue #1772
     "ignore:The interpolation= argument to 'quantile' is deprecated. Use 'method=' instead:DeprecationWarning",  # Issue #1772
     'ignore: Exception ignored in:pytest.PytestUnraisableExceptionWarning',  #FIXME: Exception ignored in: <_io.FileIO [closed]>
-    'ignore:invalid value encountered in true_divide:RuntimeWarning',  #FIXME
+    'ignore:invalid value encountered in (?:true_|divide):RuntimeWarning',  #FIXME
     'ignore:invalid value encountered in add:RuntimeWarning',  #FIXME
     "ignore:In future, it will be an error for 'np.bool_' scalars to be interpreted as an index:DeprecationWarning",  #FIXME: tests/test_tensor.py::test_pdf_eval[pytorch]
     'ignore:Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with:UserWarning',  #FIXME: tests/test_optim.py::test_minimize[no_grad-scipy-pytorch-no_stitch]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -83,11 +83,11 @@ filterwarnings = [
     'ignore:the `interpolation=` argument to percentile was renamed to `method=`, which has additional options:DeprecationWarning',  # Issue #1772
     "ignore:The interpolation= argument to 'quantile' is deprecated. Use 'method=' instead:DeprecationWarning",  # Issue #1772
     'ignore: Exception ignored in:pytest.PytestUnraisableExceptionWarning',  #FIXME: Exception ignored in: <_io.FileIO [closed]>
-    'ignore:invalid value encountered in *divide:RuntimeWarning',  #FIXME
+    'ignore:invalid value encountered in (true_)?divide:RuntimeWarning',  #FIXME
     'ignore:invalid value encountered in add:RuntimeWarning',  #FIXME
     "ignore:In future, it will be an error for 'np.bool_' scalars to be interpreted as an index:DeprecationWarning",  #FIXME: tests/test_tensor.py::test_pdf_eval[pytorch]
     'ignore:Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with:UserWarning',  #FIXME: tests/test_optim.py::test_minimize[no_grad-scipy-pytorch-no_stitch]
-    'ignore:divide by zero encountered in *divide:RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
+    'ignore:divide by zero encountered in (true_)?divide:RuntimeWarning',  #FIXME: pytest tests/test_tensor.py::test_pdf_calculations[numpy]
     'ignore:[A-Z]+ is deprecated and will be removed in Pillow 10:DeprecationWarning',  # keras
 ]
 

